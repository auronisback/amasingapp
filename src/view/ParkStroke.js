import React, {Component} from 'react';
import {Polyline} from "react-google-maps";

/**
 * Component used to manage park strokes.
 * */
export default class ParkStroke extends Component {

  /**
   * Renders the component.
   * @return The React Component rendered
   * */
  render() {
    return (
        <Polyline
            path={[this.props.start, this.props.end]}
            options={{
              strokeColor: this.props.color,
              strokeOpacity: this.props.opacity
            }}
        />
    );
  }
}