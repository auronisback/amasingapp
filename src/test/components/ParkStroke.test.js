import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ParkStroke from "../../components/ParkStroke";

Enzyme.configure({adapter: new Adapter()});

describe('<ParkStroke>', () => {
  it('Should render without crashing', () => {
    const stubPath = [{lat: 1, lng: 1}, {lat: 2, lng: 2}];
    const stubColor = '#FFF';
    const stubOpacity = 0.5;
    const stroke = shallow(<ParkStroke
        path={stubPath}
        color={stubColor}
        opacity={stubOpacity}
    />);
    expect(stroke).toMatchSnapshot();
  });
});