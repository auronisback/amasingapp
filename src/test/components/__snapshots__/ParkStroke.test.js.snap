// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkStroke> Should render without crashing 1`] = `
<Polyline
  options={
    Object {
      "strokeColor": "#FFF",
      "strokeOpacity": 0.5,
    }
  }
  path={
    Array [
      undefined,
      undefined,
    ]
  }
/>
`;
